x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/index.tsx

```diff
@@ -0,0 +1,617 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect, useMemo, useCallback } from 'react';
+import { useRouteMatch } from 'react-router-dom';
+import styled from 'styled-components';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiButtonEmpty,
+  EuiButton,
+  EuiSteps,
+  EuiBottomBar,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiErrorBoundary,
+} from '@elastic/eui';
+import type { EuiStepProps } from '@elastic/eui/src/components/steps/step';
+import { safeLoad } from 'js-yaml';
+
+import { useCancelAddPackagePolicy, useOnSaveNavigate } from '../hooks';
+import { dataTypes, FLEET_SYSTEM_PACKAGE, splitPkgKey } from '../../../../../../../common';
+import type {
+  AgentPolicy,
+  NewAgentPolicy,
+  NewPackagePolicy,
+  PackagePolicy,
+} from '../../../../types';
+import {
+  sendCreatePackagePolicy,
+  useStartServices,
+  useConfig,
+  sendGetAgentStatus,
+  useGetPackageInfoByKey,
+  sendCreateAgentPolicy,
+} from '../../../../hooks';
+import { Loading, Error, ExtensionWrapper } from '../../../../components';
+
+import { agentPolicyFormValidation, ConfirmDeployAgentPolicyModal } from '../../components';
+import { useUIExtension } from '../../../../hooks';
+import type { PackagePolicyEditExtensionComponentProps } from '../../../../types';
+import { pkgKeyFromPackageInfo } from '../../../../services';
+
+import type {
+  PackagePolicyFormState,
+  AddToPolicyParams,
+  CreatePackagePolicyParams,
+} from '../types';
+
+import { IntegrationBreadcrumb } from '../components';
+
+import { CreatePackagePolicySinglePageLayout, PostInstallAddAgentModal } from './components';
+import type { PackagePolicyValidationResults } from './services';
+import { validatePackagePolicy, validationHasErrors } from './services';
+import { StepConfigurePackagePolicy } from './step_configure_package';
+import { StepDefinePackagePolicy } from './step_define_package_policy';
+import { SelectedPolicyTab, StepSelectHosts } from './step_select_hosts';
+
+const StepsWithLessPadding = styled(EuiSteps)`
+  .euiStep__content {
+    padding-bottom: ${(props) => props.theme.eui.paddingSizes.m};
+  }
+
+  // compensating for EuiBottomBar hiding the content
+  @media (max-width: ${(props) => props.theme.eui.euiBreakpoints.m}) {
+    margin-bottom: 100px;
+  }
+`;
+
+const CustomEuiBottomBar = styled(EuiBottomBar)`
+  /* A relatively _low_ z-index value here to account for EuiComboBox popover that might appear under the bottom bar */
+  z-index: 50;
+`;
+
+export const CreatePackagePolicySinglePage: CreatePackagePolicyParams = ({
```

Comment:
The hooks and callbacks in this component are pretty complicated, is this mostly copy/moved from the previous implementation?

Reply:
Yeah its all copied, the logic is ocmplicated because we have quite a few different behaviours once the integration is installed, e.g showing modal/tutorial